src/ls/Git_helper.ml

```diff
@@ -0,0 +1,81 @@
+(*diff unified format regex https://www.gnu.org/software/diffutils/manual/html_node/Detailed-Unified.html#Detailed-Unified
+ * The above documentation isn't great, so unified diff format is
+ * @@ -start,count +end,count @@
+ * where count is optional
+ * Start and end here are misnomers. Start here refers to where this line starts in the A file being compared
+ * End refers to where this line starts in the B file being compared
+ * So if we have a line that starts at line 10 in the A file, and starts at line 20 in the B file, then we have
+ * @@ -10 +20 @@
+ * If we have a multiline diff, then we have
+ * @@ -10,3 +20,3 @@
+ * where the 3 is the number of lines that were changed
+ * We use a named capture group for the lines, and then split on the comma if it's a multiline diff *)
+let _git_diff_lines_re = {|@@ -\d*,?\d* \+(?P<lines>\d*,?\d*) @@|}
+let git_diff_lines_re = SPcre.regexp _git_diff_lines_re
+
+let is_git_repo () =
+  let cmd = ("git", [| "git"; "rev-parse"; "--is-inside-work-tree" |]) in
+  (* We use Lwt_process.exec here instead of the standard library because we can do timeouts *)
+  let%lwt status =
```

Comment:
@dinosaure started to use the Bos library for such things. We should probably unify things.

Reply:
I think it's indeed the right choice & allow us to be outside the `lwt` monad.